Compatibility issue: Proxy mode conflicts with the new version URL path `/v1/rerank` and cannot be used

## Routine Check

- [x] I have confirmed there are no similar issues
- [x] I have confirmed I am using the latest version
- [x] I have thoroughly read the project README
- [x] I understand and am willing to follow up on this issue, assist with testing and provide feedback

## Issue Description
We encountered the same problem as https://github.com/songquanpeng/one-api/issues/1604. We are glad that the author merged some important PRs to solve the problem. However, we found that the latest image is not compatible with `proxy`.

Due to the major introduction of https://github.com/songquanpeng/one-api/pull/2195, `oneapi` can naturally support the `rank` model, which is very good. However, before it appeared, users usually used the `proxy` mode to indirectly implement the use of the rank model. Since the proxy URL is generally `http://127.0.0.1:3000/v1/oneapi/proxy/{channel_id}/v1/rerank`, where `/v1/rerank` is the corresponding forwarded URL, it conflicts with the URL rules of https://github.com/songquanpeng/one-api/pull/2195 (the URL contains the string `rerank`), making it impossible to continue to use `proxy` with backward compatibility.

## Steps to Reproduce

```
curl --location 'http://127.0.0.1:3000/v1/oneapi/proxy/{channel_id}/v1/rerank' \
--header 'Authorization: {{key}}' \
--header 'Content-Type: application/json' \
--data '{
    "model": "anchor-model",
    "return_documents": true,
    "query": "hi",
    "documents": [
        "hello"
    ]
}'
```

```
[ERROR] 2025/04/09 - 15:07:51 | relay/adaptor/openai/util.go:13 [ErrorWrapper] [convert_request_failed]notimplement
github.com/songquanpeng/one-api/relay/adaptor/proxy.(*Adaptor).ConvertRequest
        /workspace/one-api/relay/adaptor/proxy/adaptor.go:28
github.com/songquanpeng/one-api/relay/controller.getRequestBody
        /workspace/one-api/relay/controller/text.go:136
github.com/songquanpeng/one-api/relay/controller.RelayTextHelper
        /workspace/one-api/relay/controller/text.go:71
github.com/songquanpeng/one-api/controller.relayHelper
        /workspace/one-api/controller/relay.go:42
github.com/songquanpeng/one-api/controller.Relay
        /workspace/one-api/controller/relay.go:52
github.com/gin-gonic/gin.(*Context).Next
        /data/gopath/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185
github.com/songquanpeng/one-api/router.SetRelayRouter.Distribute.func5
```

## Expected Behavior

Maintain backward compatibility

--- a/relay/relaymode/helper.go
+++ b/relay/relaymode/helper.go
@@ -4,7 +4,9 @@ import "strings"
 
 func GetByPath(path string) int {
 	relayMode := Unknown
-	if strings.HasPrefix(path, "/v1/chat/completions") {
+	if strings.HasPrefix(path, "/v1/oneapi/proxy") {
+		relayMode = Proxy
+	} else if strings.HasPrefix(path, "/v1/chat/completions") {
 		relayMode = ChatCompletions
 	} else if strings.HasPrefix(path, "/v1/completions") {
 		relayMode = Completions
@@ -32,8 +34,6 @@ func GetByPath(path string) int {
 		relayMode = AudioTranslation
 	} else if strings.HasPrefix(path, "/v1/images/edits") {
 		relayMode = ImagesEdits
-	} else if strings.HasPrefix(path, "/v1/oneapi/proxy") {
-		relayMode = Proxy
 	}
 
 	return relayMode

